app/src/main/java/org/walleth/functions/Formatting.kt

```diff
@@ -3,15 +3,23 @@ package org.walleth.functions
 import org.walleth.data.tokens.Token
 import java.math.BigDecimal
 import java.math.BigInteger
-import java.math.MathContext
-import java.math.RoundingMode
+import java.text.DecimalFormat
+
+val formatter: DecimalFormat
+    get() = DecimalFormat("#.#####")
+val roundingFormatter: DecimalFormat
+    get() = DecimalFormat("~#.#####")
 
 fun Token.decimalsInZeroes() = "0".repeat (decimals)
 fun Token.decimalsAsMultiplicator() = BigDecimal("1"+this.decimalsInZeroes())
 
 fun BigInteger.toValueString(token: Token) = BigDecimal(this).toValueString(token)
 
 fun BigDecimal.toValueString(token: Token): String {
-    val inEther = divide(BigDecimal("1" + token.decimalsInZeroes()))
-    return inEther.round(MathContext(6, RoundingMode.FLOOR)).stripTrailingZeros().toPlainString()
-}
\ No newline at end of file
+    val inEther = divide(BigDecimal("1" + token.decimalsInZeroes())).stripTrailingZeros()
+    if (inEther.scale() < 6) {
```

Comment:
Not sure if we really need 2 formatters here - all we want is to prefix a ~ 
So I tend more towards:
val formatted = formatter.format(inEther)
return if (inEther.scale() < 6)  formatted else "~$formatted"